### PR TITLE
feat(shorebird_code_push_protocol): add canSideload field to CreateReleaseArtifactRequest and add support to CLI

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds the `canSideload` to CreateReleaseArtifactRequest to indicate whether the artifact can be downloaded and run using `shorebird preview`.

Part of https://github.com/shorebirdtech/shorebird/issues/981

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_code_push_protocol/lib/src/messages/create_release_artifact/create_release_artifact_request.dart [hunk 1]
```diff
@@ -32,11 +36,19 @@ class CreateReleaseArtifactRequest {
   /// The hash of the artifact.
   final String hash;
 
+  /// Whether the artifact can installed and run on a device/emulator as-is.
+  @JsonKey(fromJson: _parseStringToBool, toJson: _parseBoolToString)
```
Comment (felangel) [inline packages/shorebird_code_push_protocol/lib/src/messages/create_release_artifact/create_release_artifact_request.dart:40]: why is this needed? json_serializable can handle bools afaik why do we need to send it down as a string?

Comment (bryanoltman) [reply packages/shorebird_code_push_protocol/lib/src/messages/create_release_artifact/create_release_artifact_request.dart:40]: I left a note in the doc comment for the class – this type is used to populate the `fields` of a http.MultipartRequest, which is `Map<String, String>`, so all fields serialize to a string. lmk if that could be clearer.

Comment (bryanoltman) [reply packages/shorebird_code_push_protocol/lib/src/messages/create_release_artifact/create_release_artifact_request.dart:40]: The other part of this change was to `createReleaseArtifact` in `CodePushClient`

**Review by felangel — APPROVED** [review]
